Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -34,6 +54,39 @@ public class ResourceDictionary : IResourceDictionary, IDictionary<string, objec
 		}
 
 		ResourceDictionary _mergedInstance;
+		ICollection<ResourceDictionary> _mergedDictionaries;
+		public ICollection<ResourceDictionary> MergedDictionaries { get { return _mergedDictionaries; } }
+
+		public ResourceDictionary()
+		{
+			var collection = new ObservableCollection<ResourceDictionary>();
+			collection.CollectionChanged += MergedDictionaries_CollectionChanged;
+			_mergedDictionaries = collection;
+		}
+
+		private void MergedDictionaries_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+		{
+			// New Items
+			var newItems = e.NewItems?.Cast<ResourceDictionary>();
+			if (newItems != null)
+			{
+				foreach (var item in newItems)
+					item.ValuesChanged += Item_ValuesChanged;
+
+				if (newItems.Count() > 0)
+					OnValuesChanged(newItems.SelectMany(x => x).ToArray());
+			}
+			// Old Items
+			var oldItems = e.OldItems?.Cast<ResourceDictionary>();
+			if (oldItems != null)
+				foreach (var item in oldItems)
+					item.ValuesChanged -= Item_ValuesChanged;
```

Comment:
@StephaneDelcroix - any preference on the options to take? I think the option of a new ObservableCollection, overriding ClearItems will be the best, in this scenario. It's the least complex, has a very minor perf hit, but better than taking a memory hit, considering its unlikely Clear would be called on this list.